R/resultAnnotations.R

```diff
@@ -0,0 +1,925 @@
+    # seqlevelsStyle(fds) <- seqlevelsStyle(txdb)[1]
+    fds_known <- fds[unique(to(findOverlaps(introns_tmp, 
+                                            rowRanges(fds, type = "j"), 
+                                            type = "equal"))),]
+    grIntrons <- rowRanges(fds_known, type="j")
+    introns <- as.data.table(grIntrons)
+    introns <- introns[,.(seqnames, start, end, strand)]
+    
+    sampleCounts <- K(fds_known, type = "j")
+    introns[, "meanCount" := rowMeans(sampleCounts)]
+    introns[, "medianCount" := rowMedians(as.matrix(sampleCounts))]
+    intron_ranges <- makeGRangesFromDataFrame(introns, 
+                                                keep.extra.columns = TRUE)
+    
+    # prepare the results column
+    junctions_dt[, spliceType := "complex"]
+    junctions_dt[, causesFrameshift := "inconclusive"]
+    junctions_dt[annotatedJunction == "both" & deltaPsi >= 0, 
+                    spliceType := "annotatedIntron_increasedUsage"]
+    junctions_dt[annotatedJunction == "both" & deltaPsi < 0, 
+                    spliceType := "annotatedIntron_reducedUsage"]
+    junctions_dt[annotatedJunction == "both", causesFrameshift := "unlikely"]
+    
+    # TODO check for intron retention
+    
+    starts <- which(junctions_dt[psi_positions]$annotatedJunction=="start")
+    ends <- which(junctions_dt[psi_positions]$annotatedJunction=="end")
+    nones <- which(junctions_dt[psi_positions]$annotatedJunction=="none")
+    
+    print("calculating aberrant splice types")
+    print("start junctions")
+    start_results <- sapply(starts, function(i){
+        # find the most freq intron that overlaps again
+        overlap <- to(findOverlaps(junctions_gr[i], intron_ranges, 
+                                    type = "start"))
+        expre <- sapply(overlap, function(j){
+            elementMetadata(intron_ranges[j])$medianCount
+        })
+        maxExpr <- which.max(expre)
+        return(compareEnds(junctions_gr, i, overlap[maxExpr], F, 
+                            intron_ranges, exons))
+    })
+    junctions_dt[psi_positions[starts], 
+                    causesFrameshift:=start_results[2,]]
+    junctions_dt[psi_positions[starts], 
+                    spliceType := start_results[1,]]
+    
+    print("end junctions")
+    end_results <- sapply(ends, function(i){
+        # find the most freq intron that overlaps again
+        overlap <- to(findOverlaps(junctions_gr[i], intron_ranges, 
+                                    type = "end"))
+        expre <- sapply(overlap, function(j){
+            elementMetadata(intron_ranges[j])$medianCount
+        })
+        maxExpr <- which.max(expre)
+        return(compareStarts(junctions_gr, i, overlap[maxExpr], F, 
+                                intron_ranges, exons))
+        
+    })
+    junctions_dt[psi_positions[ends], causesFrameshift:=end_results[2,]]
+    junctions_dt[psi_positions[ends], spliceType := end_results[1,]]
+    
+    print("none junctions pt1")
```

Comment:
what does pt1 and pt2 means? This is unclear to the user

Reply:
I assume they stand for "part1" and "part2". But in the end I think we can also remove those prints throughout the whole function entirely or at least reduce them strongly, as I think they are not very informative to the user

Reply:
Reducing it to a minimum is also fine with me.